### PR TITLE
Issue #2716 show usage if hostfolder name is empty

### LINKED ISSUE #2716: Adding of hostfolder through non interactive mode does not throw any error if name field is blank
### General information

  * Minishift version: master
  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

### Steps to reproduce

  1. Add hostfolder through non interactive mode
```
$ ./minishift hostfolder add -t sshfs --source /Users/gotest --target /home/shared-directory ""
$ ./minishift hostfolder add -t sshfs --source /Users/gotest --target /home/shared-directory " "
$ ./minishift hostfolder list
Name        Type    Source          Mountpoint               Mounted
shareName   sshfs   /Users/gotest   /home/shared-directory   N
            sshfs   /Users/gotest   /home/shared-directory   N
            sshfs   /Users/gotest   /home/shared-directory   N
```
  2. 
  3. 
  4. 

### Expected
Should throw error like it does in interactive mode 
```
$ ./minishift hostfolder add --interactive
Name: 
you need to specify a name
$
```

### Actual
No error thrown 

### Logs

You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #2716 

Steps to test the pull request

  1. `minishift hostfolder add --type sshfs --source /tmp/foo --target /mnt/sda1/foo " "`

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
The CI failure is not related to the PR. https://ci.centos.org/job/minishift-pr/4803/consoleFull

**Comment by LalatenduMohanty** [comment]
retest this please

File: cmd/minishift/cmd/hostfolder/add_test.go [hunk 1]
```diff
@@ -42,6 +42,21 @@ func Test_host_folder_name_required(t *testing.T) {
 	addHostFolder(nil, nil)
 }
 
+func Test_host_folder_name_not_empty(t *testing.T) {
```
Comment (amitkrout) [inline cmd/minishift/cmd/hostfolder/add_test.go]: @anjannath Some set of lines are common in all the Test_ function. Can we keep it separate function and make a call when it is needed ?
```
var err error
tmpMinishiftHomeDir := cli.SetupTmpMinishiftHome(t)	config.InstanceConfig, err = config.NewInstanceConfig(filepath.Join(tmpMinishiftHomeDir, "config"))
config.AllInstancesConfig, err = config.NewAllInstancesConfig(filepath.Join(tmpMinishiftHomeDir, "config"))	assert.NoError(t, err, "Unexpected error setting instance config")

tee := cli.CreateTee(t, true)
defer cli.TearDown(tmpMinishiftHomeDir, tee)
defer viper.Reset()
```

**Comment by amitkrout** [comment]
@LalatenduMohanty This is happening due to inconsistency in failure of ```oc clulter up``` in CI for okd 3.10. I am putting some ```oc cluster up``` failure collected form ```m start``` log

```
10:58:24 E0823 06:58:23.615839    2018 interface.go:34] Failed to install "kube-proxy": failed to install "kube-proxy": Docker run error rc=1; caused by: Docker run error rc=1 
```
```
06:53:07 E0823 02:53:07.021717    1991 interface.go:34] Failed to install "openshift-web-console-operator": Operation cannot be fulfilled on openshiftwebconsoleconfigs.webconsole.operator.openshift.io "instance": the object has been modified; please apply your changes to the latest version and try again
```

There is one mismatch between ``FeatureContext``` and argument send from basic.feature file line no [39](https://github.com/minishift/minishift/blob/master/test/integration/features/basic.feature#L39). If ```s.Step``` in ```FeatureContext``` mismatch with argument send from scenario steps then godog does not report any failure, returns step is not defined instead. So probably it was not noticed while merging #2702

Just replace ```1000ms``` instead of ```1s``` will fix the problem cc_ @anjannath

File: cmd/minishift/cmd/hostfolder/add.go [hunk 2]
```diff
@@ -99,8 +99,11 @@ func addHostFolder(cmd *cobra.Command, args []string) {
 		}
 		if interactive {
 			name, shareType = readNameAndTypeInteractive()
+		}
+		if len(strings.TrimSpace(args[0])) == 0 {
+			atexit.ExitWithMessage(1, usage)
 		} else {
-			name = args[0]
+			name = strings.TrimSpace(args[0])
```
Comment (LalatenduMohanty) [inline cmd/minishift/cmd/hostfolder/add.go]: @anjannath I suggest to modify the code as below to make it more readable. WDYT?

```diff
diff --git a/cmd/minishift/cmd/hostfolder/add.go b/cmd/minishift/cmd/hostfolder/add.go
index a6eed6b9..21455acc 100644
--- a/cmd/minishift/cmd/hostfolder/add.go
+++ b/cmd/minishift/cmd/hostfolder/add.go
@@ -99,11 +99,11 @@ func addHostFolder(cmd *cobra.Command, args []string) {
                }
                if interactive {
                        name, shareType = readNameAndTypeInteractive()
-               }
-               if len(strings.TrimSpace(args[0])) == 0 {
-                       atexit.ExitWithMessage(1, usage)
                } else {
                        name = strings.TrimSpace(args[0])
+                       if len(strings.TrimSpace(name)) == 0 {
+                               atexit.ExitWithMessage(1, usage)
+                       }
                }
        }
```

code snippet looks like this 

```
if usersShare && runtime.GOOS == "windows" {
		// Windows-only (CIFS), all instances
		name = "Users"
	} else {
		if len(args) < 1 && !interactive {
			atexit.ExitWithMessage(1, usage)
		}
		if interactive {
			name, shareType = readNameAndTypeInteractive()
		} else {
			name = strings.TrimSpace(args[0])
			if len(strings.TrimSpace(name)) == 0 {
				atexit.ExitWithMessage(1, usage)
			}
		}
	}
```

Comment (anjannath) [reply cmd/minishift/cmd/hostfolder/add.go]: @LalatenduMohanty Looks good!

File: cmd/minishift/cmd/hostfolder/add_test.go (hunk 1 above)
Comment (anjannath) [reply cmd/minishift/cmd/hostfolder/add_test.go]: @amitkrout I Think this should be done as part of a refactor, a separate commit. WDYT? This PR is only for the fix about blank names.

Comment (anjannath) [reply cmd/minishift/cmd/hostfolder/add_test.go]: Or i can add a separate commit for removing the duplications

Comment (amitkrout) [reply cmd/minishift/cmd/hostfolder/add_test.go]: > I Think this should be done as part of a refactor

+1, lets keep it in a separate pr. Please create another issue something like ```refactor unit test to remove duplicate lines```

**Comment by anjannath** [comment]
retest this please

File: cmd/minishift/cmd/hostfolder/add_test.go (hunk 1 above)
Comment (LalatenduMohanty) [reply cmd/minishift/cmd/hostfolder/add_test.go]: > Please create another issue something like refactor unit test code to remove duplicate lines

+1
